spec/support/asn1_helper.rb

```diff
@@ -0,0 +1,11 @@
+module Asn1Helper
+  def asn1_dig(obj)
+    if obj.respond_to?(:first) && obj.first.is_a?(Enumerable)
+      asn1_dig obj.first
+    elsif obj.respond_to?(:value) && obj.value.is_a?(Enumerable)
+      asn1_dig obj.value
+    else
+      obj
+    end
```

Comment:
Yeah, it's definitely more clear but might break again somewhere in the future. Then again, ASN1 is always a PITA to work with so let's deal with it when the problem arises 😀